### PR TITLE
*: audit operation names in new schema changer

### LINKED ISSUE #84583: scop: audit operation names
Declarative schema changer operation names and their field names are mildly inconsistent. For instance we have a pattern where index state changing ops are named `MakeBackfilledIndexMerging`, `MakeBackfillingIndexDeleteOnly`, etc. but we also have `MakeAddedIndexDeleteAndWriteOnly` which doesn't conform to the pattern. As far as fields are concerned we have a mix of `DescID`, `DescriptorID` etc. 

None of this matters in any other way than: this stuff shows up in `EXPLAIN (DDL)` output. Let's make it consistent in time for this release. It shouldn't be hard or take long, but will result in noisy test changes.

Jira issue: CRDB-17746

### PR DESCRIPTION
**Pull request by Xiang-Gu**
This PR made non-functional changes related to new schema changer ops:
1. Rename all status changing operation on Index and Column to comform
to the pattern: `Make[StatusA][Index/Column][StatusB]`

2. Rename a few constant/field/function names to be consistent with the
principle of "being explicit":
  2.1. Rename operation name `XxxGcXxx` to `XxxGCXxx`;
  2.2. Rename field name in operations `DescID` to `DescriptorID`;
  2.3. Rename descpb.DELETE_AND_WRITE_ONLY to descpb.WRITE_ONLY;
  2.4. Rename sql.RunningStatusDeleteAndWriteOnly to sql.RunningStatusWriteOnly

3. Seveal comments enhencement

4. Rename `DELETE_AND_WRITE_ONLY` to `WRITE_ONLY` in some files under
docs/.

Fixes #84583

Release note: None

### CODE REVIEW

**Comment by postamar** [comment]
I like all of this, except 2.3. The choice of `WRITE_ONLY` was deliberate and ideally we'd be renaming `descpb.DELETE_AND_WRITE_ONLY` instead.

File: pkg/sql/schema_changer.go [hunk 1]
```diff
@@ -82,10 +82,10 @@ const (
 	// the cluster to converge to seeing the schema element in the DELETE_ONLY
 	// state.
 	RunningStatusDeleteOnly jobs.RunningStatus = "waiting in DELETE-ONLY"
-	// RunningStatusDeleteAndWriteOnly is for jobs that are currently waiting on
+	// RunningStatusWriteOnly is for jobs that are currently waiting on
 	// the cluster to converge to seeing the schema element in the
-	// DELETE_AND_WRITE_ONLY state.
-	RunningStatusDeleteAndWriteOnly jobs.RunningStatus = "waiting in DELETE-AND-WRITE_ONLY"
+	// WRITE_ONLY state.
+	RunningStatusWriteOnly jobs.RunningStatus = "waiting in WRITE_ONLY"
```
Comment (postamar) [inline pkg/sql/schema_changer.go:88]: @ajwerner is it OK to change these running status messages? I think so but I ask just to be sure.

File: pkg/sql/schema_changer.go [hunk 2]
```diff
@@ -2760,7 +2760,7 @@ func (r schemaChangeResumer) OnFailOrCancel(
 		}
 	}
 
-	// If this is a schema change to drop a database or schema, DescID will be
+	// If this is a schema change to drop a database or schema, DescriptorID will be
```
Comment (postamar) [inline pkg/sql/schema_changer.go:2763]: nit: this change should not be made

**Review by postamar — APPROVED** [review]
LGTM, thanks for doing this, assuming that changing the running status text is OK.

File: pkg/sql/schema_changer.go (hunk 2 above)
Comment (Xiang-Gu) [reply pkg/sql/schema_changer.go:2763]: done

File: pkg/sql/schema_changer.go (hunk 1 above)
Comment (Xiang-Gu) [reply pkg/sql/schema_changer.go:88]: andrew said to me offline that this is fine.

**Comment by Xiang-Gu** [comment]
TFTR!

bors r+

**Pull request merged**